### DISCUSSION TITLE
Possible configuration issue due to updated packages

### DISCUSSION THREAD

**Discussion by kcphila** [apache/airflow#24434] [General]
### Apache Airflow version

2.3.2 (latest released)

### What happened

This seems odd and there may be another answer for it, but I can't find anything. We have been running 2.1.4 on Ubuntu Xenial for about a year without much trouble. So, we are switching over to the new Ubuntu 22.04 LTS environment with all up to date libraries.

When attempting to run Airflow 2.3.2, I am getting strange config-related errors when processing our custom config variable (full stack trace at the bottom).

1. ` log_processor_filename_template = {{{{ filename }}}}.log` - This is the value included in the default template, but it fails to process with the error `jinja2.exceptions.TemplateSyntaxError: expected token ':', got '}'`
2. The default `log_filename_template` value also fails to process with a similar error

Commenting these out allows the config file to parse, but then:

3. The config is not substituting template values correctly.  The multiple `{AIRFLOW_HOME}` references are being interpreted literally instead of being substituted for the actual value. If I change these to reference an environment variable (e.g., `${AIRFLOW_HOME}`), it works fine. 

Here's the stack trace for the log filename template. I put in a custom breakpoint to determine that it indeed is just for the log_filename_template and log_processor_filename_template variables:

```
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: Unable to load the config, contains a configuration error.
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: Traceback (most recent call last):
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3.10/logging/config.py", line 565, in configure
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     handler = self.configure_handler(handlers[name])
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3.10/logging/config.py", line 746, in configure_handler
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     result = factory(**kwargs)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/utils/log/file_processor_handler.py", line 44, in _init_
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     self.filename_template, self.filename_jinja_template = parse_template_string(filename_template)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/utils/helpers.py", line 176, in parse_template_string
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     return None, jinja2.Template(template_string)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1195, in _new_
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     return env.from_string(source, template_class=cls)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1092, in from_string
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     return cls.from_code(self, self.compile(source), gs, None)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 757, in compile
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     self.handle_exception(source=source_hint)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 925, in handle_exception
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     raise rewrite_traceback_stack(source=source)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "<unknown>", line 1, in template
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: jinja2.exceptions.TemplateSyntaxError: expected token ':', got '}'
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: The above exception was the direct cause of the following exception:
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: Traceback (most recent call last):
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/bin/airflow", line 5, in <module>
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     from airflow._main_ import main
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/_init_.py", line 47, in <module>
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     settings.initialize()
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/settings.py", line 561, in initialize
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     LOGGING_CLASS_PATH = configure_logging()
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/logging_config.py", line 73, in configure_logging
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     raise e
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/local/lib/python3.10/dist-packages/airflow/logging_config.py", line 68, in configure_logging
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     dictConfig(logging_config)
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3.10/logging/config.py", line 810, in dictConfig
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     dictConfigClass(config).configure()
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:   File "/usr/lib/python3.10/logging/config.py", line 572, in configure
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]:     raise ValueError('Unable to configure handler '
Jun 13 18:05:08 ip-10-30-132-24 airflow[196893]: ValueError: Unable to configure handler 'processor'
```

### What you think should happen instead

Whatever is in the [official default template](https://github.com/apache/airflow/blob/2.3.2/airflow/config_templates/default_airflow.cfg) should work. I don't mind using the bash syntax in the config file, but this seems very strange.

### Operating System

Ubuntu 22.04 AMD64 running on AWS

### Versions of Apache Airflow Providers

Airflow 2.3.2 

### Deployment

Other

### Deployment details

pip-based install. I've tried this with jinja2 versions 3.1 and 3.0.3

### Are you willing to submit PR?

If I can identify a code-based issue, sure.

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by uranusjr** [comment]
The default config template is itself a template, so `{{{{ filename }}}}.log` actually renders to `{{ filename }}.log` at runtime. This is probably the cause of your problem.

**Comment by potiuk** [comment]
Converted to disussion - until the user confirms. And yeah this is the likely reason @kcphila - maybe (if you confirm this is the right advice) you could provide a PR/doc update to warn users against similar problems (I think you are the best person to know where you looked for help and did not find it).

**Comment by kcphila** [comment]
@potiuk - I'll verify shortly, but this sounds accurate. I was doing a manual merge to determine what fields and values had been updated in the config between versions to keep the rest of our custom parameters aligned. 

And I'll be happy to submit a PR with suggestions.

**Comment by kcphila** [comment]
This indeed looks like the issue.  Thank you for the quick comments. I'll look up the contributor guidelines and think about how to formulate it so that future users hopefully find the notes before they get too bogged down into details.  Have now looked at the source code, I think there are other comments that should be included - like that the command-based configuration options are actually using `shlex` to use a safe-ish subprocess command, which will also silently break intuitive attempts to write a bash command (i.e. ```echo  postgresql+psycopg2://user:`cat /path/to/securefile`@db/airflow_db"```)


I do have an additional question, in [configuration.py](https://github.com/apache/airflow/blob/v2-3-stable/airflow/configuration.py), `expand_env_var()` is used to interpolate environment variables into the values of the config file often, but not always. It's used when the configuration value is set in the environment (line 447), when reading in default values (line 546), and for regular config variables defined in the config file (593). But, it is not used when either the environment (line 453) or the configuration variable (line 467, 483) specifies a command to run.  Is there a specific reason why the commands do not use environment variable expansion but everything else does?

The result is that one can do this, which is obviously not ideal:
```
sql_alchemy_conn= postgresql+psycopg2://${DATABASE_USER}:${SECRET_EXPOSED_IN_ENVIRONMENT}@${DATABASE_HOST}/${AIRFLOW_DATABASE}"
```
But not this, which at least can take advantage of OS permissions to limit threats:
```
sql_alchemy_conn_cmd = "${AIRFLOW_HOME}/script/secure_script_to_generate_postgres_connection_string"
```

**Reply by potiuk** [reply]
I think expand_env_var is a bit dangerous when ther eis a command to run - because you could basically change what's run via env var. I am not sure if it was the root cause, but that what I think was the original idea.

**Reply by kcphila** [reply]
It makes sense that it was the original idea, but I wonder if it's more limiting than helpful here. Because the command is being called with the list-based `Popen`, command args are santizied to prevent shell expansion, which adds a lot of the desired protection. And if one can already change environment variables, they can just change `AIRFLOW__DATABASE__SQL_ALCHEMY_CONN_CMD` to point to a shell script they write in `/tmp` to enact their nefarious ideas, and so preventing environment variables in the `*_cmd` keys seems like it won't actually prevent bad actors.

But since the sql_alchemy_conn_cmd can't access the core / root env vars like `${AIRFLOW_HOME}`, it seems to make them of very limited use unless you want to set an absolute path to a global shell script, which doesn't seem would fit well into most people's environments.

**Reply by potiuk** [reply]
Not necessarily. Depending on your circumstances you might, or might not be able to edit various variables. For exmple you might not be able to configure AIRFLOW__ commands as they might be set in environment and "hard-coded" but you might influence the name of the user airlfow is run with and (HOME) variable - and if you - for example set your user name to `; rm -rf /`. If the _CMD contains ${HOME} or even `~` which is not properly escaped, you would get classic `bash injection" here.

So I think the decision about preventive disabling the expansion here is a very sound one. As usual - when you are unsure if there are other similar cases, you should block the whole thing. Security trumps convenience.

**Reply by kcphila** [reply]
Hi Jarek,

That's exactly what I'm trying to get at - your example does not work because the code is already protecting against shell injection.

Looking at [configuration.py](https://github.com/apache/airflow/blob/v2-3-stable/airflow/configuration.py):

`expand_env_var()` (lines 82-96) uses `os.path.expandvars()` and `os.path.expandhome()` to, potentially recursively, expand all environment variables in code, and then `run_command()` (lines 98-111) splits the command and passes it to `subprocess.Popen` in list format. When called in list format (and without the `shell = True` param), Popen fully sanitizes the input, and so your classic injection example is protected against - that's what I'm trying to say.

There are certain times in which someone with enough access could do some bad things - but I don't see much of a scenario in which someone couldn't redefine `AIRFLOW__DATABASE__SQL_ALCHEMY_CONN_CMD` to whatever they want and could still exploit the other environment variables.

**Reply by potiuk** [reply]
When in doubt (and we both have them - you say you cannot imagine, but you will not guarantee it 100% either) - I errr on the side of caution. Marginal usablitiy improvements for very small group of people are not worth the risk even if there is a small chnce we overlooked something. Taking any risk here makes completely no sense for us as product maintainers, even if it might make sense for you as a user who wants to use this - generally - obscure - feature.

I'd say it's no go - but  - as usual in open-source, "talking the talk" has to end at some point and "walking the walk" should take over. If you are absolutely convinced that this is worth it, make a PR implementing it and fight for it in the PR, where you will be able to show code and explain all the protections to those who will be reviewing. This is how it works here.

**Reply by kcphila** [reply]
That makes sense. I might end up with a PR for more dev discussion around it... but there are more pressing concerns.  

Thanks for your guidance and entertaining my discussion.

Best,

Kevin

**Reply by krcrouse** [reply]
@potiuk , 

On the topic of contributing to the development and documentation - I've been trying to set up a dev environment to run all of the tests using breeze (followings [this doc](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst), but cannot actually spin it up. 

When attempting to run `breeze --python 3.9 --backend postgres`, it fails, saying `No such image: ghcr.io/apache/airflow/main/ci/python3.9`, but in the log it also says :

```
ERROR: Cannot install apache-airflow[devel-ci]==2.4.0.dev0 because these package versions have conflicting dependencies.
#0 4.572 
#0 4.572 The conflict is caused by:
#0 4.572     apache-airflow[devel-ci] 2.4.0.dev0 depends on flask<2.0 and >=1.1.0
#0 4.572     The user requested (constraint) flask==2.1.2
```

So,  given how this discussion began - am I missing a key step that isn't in the documentation? I initially set up a virtual environment for python 3.9, installed pipx, installed breeze, and then attempted to run it. In the virtualenv, flask is version `1.1.2` and so the conflict is not using the correct set of libraries. The flask with system python (which is 3.10) is 2.1.2, so I'm guessing breeze is either installing its own python or using system python instead of the virtualenv, even though I am specifying the python version in the command.

So, any thoughts would be appreciated.

**Reply by potiuk** [reply]
You need to rebase to latest `main`. Breeze is a development feature. It expects you to be on the latest "development" version. but you are likely few 10s or 100s commits behind.

**Comment by zsdyx** [comment]
Hi, I also met it. Is there a way to solve it

**Reply by potiuk** [reply]
Unless you open your issue with your details, and your error, commenting "I have the same problem" is useless and will not help you to solve your problem - if you can't find solution by reading the whole thread here and figuring out what to do, the only way you can reach out for help is to open a new issue and provide all the details, so that anyone who (in their free time) would lke to help you enough evidences that they can help you. Your issue might, or might not be the same (even if error is similar). By (lazily) not providing any details you close a possibillty that someone will actually decide to spend some time on it, ask more details and you getting help - it's extremely unlikely to happen.

**Comment by EugeneChung** [comment]
## ENV
- Python 3.9.14
```
python3.9 -m pip list |grep Jinja2
Jinja2                                   3.1.2
```
```
python3.9 -m pip list |grep airflow
apache-airflow                           2.4.1
```

## log_filename_template
https://github.com/apache/airflow/blob/7b979def75923ba28dd64e31e613043d29f34fce/airflow/config_templates/default_airflow.cfg#L381-L381

## error stack
```py
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]: [2022-10-12 13:38:11,875] {local_executor.py:130} ERROR - Failed to execute task expected token ':', got '}'.
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]: Traceback (most recent call last):
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/executors/local_executor.py", line 126, in _execute_work_in_fork
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     args.func(args)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/cli/cli_parser.py", line 52, in command
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return func(*args, **kwargs)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/cli.py", line 99, in wrapper
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return f(*args, **kwargs)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 380, in task_run
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     ti.init_run_context(raw=args.raw)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2503, in init_run_context
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     self._set_context(self)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/log/logging_mixin.py", line 57, in _set_context
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     set_context(self.log, context)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/log/logging_mixin.py", line 189, in set_context
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     set_context(value)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/airflow_home/config/s3_task_handler.py", line 48, in set_context
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     super().set_context(ti)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/log/file_task_handler.py", line 69, in set_context
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     local_loc = self._init_file(ti)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/airflow_home/config/file_task_handler.py", line 44, in _init_file
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return FileTaskHandler._init_file(self, ti)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/log/file_task_handler.py", line 310, in _init_file
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     relative_path = self._render_filename(ti, ti.try_number)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/log/file_task_handler.py", line 91, in _render_filename
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     str_tpl, jinja_tpl = parse_template_string(template)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/airflow/utils/helpers.py", line 168, in parse_template_string
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return None, jinja2.Template(template_string)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/jinja2/environment.py", line 1208, in __new__
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return env.from_string(source, template_class=cls)
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:   File "/home/ubuntu/.local/lib/python3.9/site-packages/jinja2/environment.py", line 1105, in from_string
Oct 12 13:38:11 ip-172-30-182-69 airflow[32494]:     return cls.from_code(self, self.compile(source), gs, None)
```

**Reply by EugeneChung** [reply]
```py
    def _render_filename(self, ti: TaskInstance, try_number: int) -> str:
        with create_session() as session:
            dag_run = ti.get_dagrun(session=session)
            template = dag_run.get_log_template(session=session).filename
            template = 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log'
        str_tpl, jinja_tpl = parse_template_string(template)
```

I set the template directly in the code, `airflow/utils/log/file_task_handler.py` and it works.
I think there is a case that the escaped delimiters`{{{{`, `}}}}`, `{{%%`, `%%}}` are not actually escaped.

**Reply by EugeneChung** [reply]
I set my log_filename_template as `log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{%% if ti.map_index >= 0 %%}map_index={{ ti.map_index }}/{%% endif %%}attempt={{ try_number }}.log`.
However, somewhere in the code changes my setting. 

```py
def parse_template_string(template_string: str) -> tuple[str | None, jinja2.Template | None]:
    """Parses Jinja template string."""
    import jinja2

    if "{{" in template_string:  # jinja mode
        print(template_string)
        return None, jinja2.Template(template_string)
    else:
        return template_string, None
```
The result of `print(template_string)` is
```
Oct 12 14:10:35 ip-172-30-182-69 airflow[36845]: dag_id={{{{ ti.dag_id }}}}/run_id={{{{ ti.run_id }}}}/task_id={{{{ ti.task_id }}}}/{{% if ti.map_index >= 0 %}}map_index={{{{ ti.map_index }}}}/{{% endif %}}attempt={{{{ try_number }}}}.log
```. 
It results in definitely the jinja2 template error.

**Reply by EugeneChung** [reply]
I printed some values at airflow.utils.db.synchronize_log_template().
It showed like below and they are okay:
```
Oct 12 23:51:56 ip-172-30-182-69 airflow[80471]: pre_upgrade_filename=dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log
Oct 12 23:51:56 ip-172-30-182-69 airflow[80471]: pre_upgrade_filename=dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log
Oct 12 23:51:56 ip-172-30-182-69 airflow[80471]: stored=LogTemplate(filename=dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log,
```

However, the value from airflow.models.dagrun.DagRun.get_log_template() is different.
```
Oct 12 23:52:28 ip-172-30-182-69 airflow[80673]: 3 <--- self.log_template_id
Oct 12 23:52:28 ip-172-30-182-69 airflow[80673]: LogTemplate(filename=dag_id={{{{ ti.dag_id }}}}/run_id={{{{ ti.run_id }}}}/task_id={{{{ ti.task_id }}}}/{{% if ti.map_index >= 0 %}}map_index={{{{ ti.map_index }}}}/{{% endif %}}attempt={{{{ try_>
Oct 12 23:52:28 ip-172-30-182-69 airflow[80673]: dag_id={{{{ ti.dag_id }}}}/run_id={{{{ ti.run_id }}}}/task_id={{{{ ti.task_id }}}}/{{% if ti.map_index >= 0 %}}map_index={{{{ ti.map_index }}}}/{{% endif %}}attempt={{{{ try_number }}}}.log
```

**Reply by potiuk** [reply]
You should not "copy" those values from the template. The "default_config" template is not there so that you copy it from there, it is there to be automatically used by airflow. And Airflow processes it via jinja so the {{{{ is replaced with {{ becasue this is how Jinja processes its templates. You shoud either generate your log configuration automatically or manually prepare it (as @EugeneChung mentioned it works for him).

**Comment by awildturtok** [comment]
Getting ` '%' must be followed by '%' or '(', found: '% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log'` when using `log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log`

using 2.5.0

**Reply by kcphila** [reply]
Hey - I believe you need to escape the % signs as well, i.e. `{%% if ti.map_index >= 0 %%}map_index={{ ti.map_index }}/{%% endif %%}`

**Reply by awildturtok** [reply]
@kcphila you are right. I think I had troubles because of the per-dag-run caching of the variable. After restarting airflow with a new server it worked.

**Reply by HansBambel** [reply]
For the sake of completion, here is the line how the default-value should be in the config:
```
log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{%% if ti.map_index >= 0 %%}map_$
```
